lib/datasource/index.ts

```diff
@@ -126,3 +130,30 @@ export function getDigest(
     value
   );
 }
+function baseUrlLegacyMassager(sourceUrl) {
+  let url: string = sourceUrl;
+  // Massage www out of github URL
+  url = url.replace('www.github.com', 'github.com');
+  if (url.startsWith('https://github.com/')) {
+    url = url
+      .split('/')
+      .slice(0, 5)
+      .join('/');
+  } // a lot of this is probably redundant and can be better achieved with URL.
+  if (url.includes('github.com')) {
```

Comment:
This if would seem to exclude github enterprise URLs from working

Reply:
GIthub enterprise?
Is that a software version of git that can be self-hosted?

Reply:
Yes, self-hosted GitHub. URLs might be anything like https://github.company.com or http://code.company.com

The current logic in the npm datasource that you're removing allows for these URLs but the new logic you added excludes them, so it would break all self-hosted GitHub Enterprise users who rely on release notes.

Reply:
Ah, I see.
Fixed, sorry about that, I have never utilized github or its products much, so I assumed that enterprise, probably meant a subdomain for your company or something similar.

Reply:
Mostly it's pretty similar to how gitlab.com vs self-hosted GitLab works